src/NUnitFramework/tests/Attributes/TestOrderAttributeTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.TestData;
+using NUnit.TestData.TestCaseSourceAttributeFixture;
+using NUnit.TestUtilities;
+
+namespace NUnit.Framework.Tests.Attributes
+{
+    [TestFixture]
+    class TestOrderAttributeTests
+    {
+        [Test]
+        public void CheckOrderIsCorrect()
+        {
+            var testFixt = TestBuilder.MakeFixture(typeof (TestCaseOrderAttributeFixture));
+            testFixt.Sort();// need to manually call this method to mimic real world conditions
```

Comment:
Just noting on this, I tried with debugging and using the test order inside a fixture in the unit test project and it worked fine, there is the method in the assembly creation that doesn't get called when you use "MakeFixture", so not sure if we should add the sort call somewhere else. I had a though to call Sort every time when a test case is added maybe?